kafka-rest/src/main/java/io/confluent/kafkarest/controllers/SchemaManagerImpl.java

```diff
@@ -26,26 +26,32 @@
 import io.confluent.kafka.schemaregistry.client.rest.exceptions.RestClientException;
 import io.confluent.kafka.serializers.subject.strategy.SubjectNameStrategy;
 import io.confluent.kafkarest.Errors;
+import io.confluent.kafkarest.config.ConfigModule.SchemaRegistryEnabledConfig;
 import io.confluent.kafkarest.entities.EmbeddedFormat;
 import io.confluent.kafkarest.entities.RegisteredSchema;
+import io.confluent.kafkarest.exceptions.BadRequestException;
 import java.io.IOException;
 import java.util.Optional;
 import javax.inject.Inject;
-import org.apache.kafka.common.errors.SerializationException;
+import org.apache.avro.SchemaParseException;
 
-final class SchemaManagerImpl implements SchemaManager {
-  private final SchemaRegistryClient schemaRegistryClient;
+public final class SchemaManagerImpl implements SchemaManager {
+  private final Optional<SchemaRegistryClient> schemaRegistryClient;
   private final SubjectNameStrategy defaultSubjectNameStrategy;
+  private final boolean schemaRegistryEnabled;
 
   @Inject
-  SchemaManagerImpl(
-      SchemaRegistryClient schemaRegistryClient, SubjectNameStrategy defaultSubjectNameStrategy) {
+  public SchemaManagerImpl(
+      Optional<SchemaRegistryClient> schemaRegistryClient,
+      SubjectNameStrategy defaultSubjectNameStrategy,
+      @SchemaRegistryEnabledConfig Boolean schemaRegistryEnabled) {
     this.schemaRegistryClient = requireNonNull(schemaRegistryClient);
     this.defaultSubjectNameStrategy = requireNonNull(defaultSubjectNameStrategy);
+    this.schemaRegistryEnabled = schemaRegistryEnabled;
   }
 
   @Override
-  public RegisteredSchema getSchema(
+  public Optional<RegisteredSchema> getSchema(
```

Comment:
Why is this `Optional`? If SR is off you throw. Otherwise you always return something. You never return `Optinal.empty`. Revert.